### PR TITLE
Views

### PR DESCRIPTION
**Pull request by tibuch**
I've wrapped the imgLib2 Views class into ops and created the ViewNamespace. 

I am not sure about the interface hierarchy. For example I have an ```Extend``` and an ```ExtendBorder``` interface. Both only extend ```Ops``` but I think ```ExtendBorder``` should extend ```Extend``` and ```Extend``` should extend ```Ops```. Is it possible to define an interface hierarchy for auto-generated interfaces?

### CODE REVIEW

**Comment by dietzc** [comment]
Hi @tibuch. Thank you very much that you are working on this. It's really great to have `Views` intergated into `Ops`. I will start reviewing (and I hope @ctrueden and @hinerm also find some time to go over it) in the next few days, starting today. So please find my comments below.

**Comment by dietzc** [comment]
Missing author information: nearly everywhere the author information + some kind of JavaDoc is missing.

**Comment by dietzc** [comment]
all of this is related to https://github.com/imagej/imagej-ops/issues/173

**Comment by tibuch** [comment]
Hi @dietzc and @ctrueden 
I rebased this branch onto master. 
Could you take a look?

**Comment by ctrueden** [comment]
Cool, thanks for the update!

As discussed, we want to categorize the ops into their appropriate namespaces, rather than having a dedicated `view` namespace.

@dietzc and I took a first crack at categorizing the view ops:

op name                        | namespace   | notes
------:                        | ---------   | -----
intervalView                   | transform   |
interpolateView                | transform   |
rasterView                     | transform   |
extendView                     | transform   |
extendMirrorSingleView         | transform   |
extendMirrorDoubleView         | transform   |
extendValueView                | transform   |
extendZeroView                 | transform   |
extendRandomView               | transform   |
extendPeriodicView             | transform   |
extendBorderView               | transform   |
rotateView                     | transform   |
permuteView                    | transform   |
translateView                  | transform   |
offsetView                     | transform   |
zeroMinView                    | transform   |
hyperSliceView                 | transform   |
addDimensionView               | transform   |
invertAxisView                 | transform   |
isZeroMinView                  | -           | Do we need this??
iterableView                   | transform   |
flatIterableView               | transform   |
collapseView                   | transform   |
realCollapseView               | transform   | rename to collapseRealView
numericCollapseView            | transform   | rename to collapseNumericView
subsampleView                  | transform   |
dropSingletonDimensionsView    | transform   |
stackView                      | transform   |
shearView                      | transform   |
unshearView                    | transform   |
permuteView                    | transform   | needs to be added still
permuteCoordinatesView         | transform   |
permuteCoordinatesInverseView  | transform   |
crop                           | transform   | migrate from image namespace
project                        | transform   | migrate from image namespace
scale                          | transform   | migrate from image namespace

In other words, we can:
* Rename the `view` namespace to `transform`.
* Migrate a couple of ops from the `image` namespace into the new `transform` namespace.

@tpietzsch Do you have opinions about the namespace classifications for these ops? We did consider alternatives such as putting the `iterableView` and `flatIterableView` into `convert`. But these are also in a sense spatial transformations.

**Comment by ctrueden** [comment]
I'll try to get this merged within the next week or so. If I fall short, @tibuch, please feel free to do it yourself once you have time.

**Comment by ctrueden** [comment]
This was a beast, but I finally got it cleaned up and merged. Pushing momentarily.
